src/Rector/AbstractDrupalCoreRector.php

```diff
@@ -29,6 +31,31 @@ public function configure(array $configuration): void
         $this->configuration = $configuration;
     }
 
+    protected function isInBackwardsCompatibleCall(Node $node): bool
+    {
+        if (!class_exists(DeprecationHelper::class)) {
+            return false;
+        }
+
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
+
+        $callStack = $scope->getFunctionCallStackWithParameters();
+        if (count($callStack) === 0) {
+            return false;
+        }
+        [$function, $parameter] = $callStack[0];
+        if (!$function instanceof MethodReflection) {
+            return false;
+        }
+        if ($function->getName() !== 'backwardsCompatibleCall'
+            || $function->getDeclaringClass()->getName() !== DeprecationHelper::class
+        ) {
+            return false;
+        }
+
+        return $parameter !== null && $parameter->getName() === 'deprecatedCallable';
+    }
```

Comment:
Does Rector have a node visitor to just stop processing so that it doesn't handle nodes in this execution tree?

Reply:
Think so yes. My old fix for phpstan would probably also work. This seemed easier is 0retty much why I used this.

Reply:
I guess we're calling it in our base class, which all rules should extend. This is fine, but could lead to a "bug" if a rule didn't. But probably a minimal problem

Reply:
valid point. But if it doesn't extend this class, it should not add the deprecation helper. But you are right though, it could slip through if we need some sort of exotic way to do things. Still would probably use the base class for the helpers.